client/src/component/Search.js

```diff
@@ -0,0 +1,45 @@
+import "./User.css";
+import { useContext } from "react";
+import { AppContext } from "../App";
+
+export function Search() {
+    const { userName, setCodewarsData, setGitHubData } = useContext(AppContext);// we suposed to use context with this name//
+    //as search button clicked this function will happen
+    const searchHandle = () => {
+        /////****GETTING CODEWARS DATA****///////
+        fetch(`https://www.codewars.com/api/v1/users/${userName}`)
+            .then((res) => {
+                if (!res.ok) {
+                    throw new Error(res.statusText);
+                }
+                return res.json();
+            })
+            .then((data) => {
+                return setCodewarsData(data);
```

Comment:
to get the codewars rank, you should follow the object keys and get the proper value, for example, data.ranks.overall.name, because we only can reach the value when we fetching and I can't do this in MilestoneTable

Reply:
thank you for this point, but in my openion we can get all the data for user so which part we need we can have access the data for user which we need it , but maybe some of data are  useless so let ask for advice in  this point

Reply:
Both valid points here, happy with what you decide.